doc/architecture-decisions/0002-add-enhanced-ecommerce.md

```diff
@@ -0,0 +1,46 @@
+# 2. Add Enhanced E-commerce plugin
+
+Date: 2017-06-27
+
+## Status
+
+Accepted
+
+## Context
+
+Rendering apps track pageviews and events to help us measure the performance
+of GOV.UK.
+
+Static uses an [abstraction layer](https://github.com/alphagov/govuk_frontend_toolkit/blob/master/docs/analytics.md) in GOV.UK Frontend Toolkit, that reports the data to Google Analytics, but the underlying provider could be changed.
+
+For analysing search performance, we want to be able to reliably track clickthrough rates on links in search results.
+
+[Enhanced E-Commerce](https://developers.google.com/analytics/devguides/collection/analyticsjs/enhanced-ecommerce) is a plugin for google analytics that can track impressions as well as clicks. We model each search result page as a "product list", and each content item as a "product".
+
+## Decision
+
+We added a plugin for E-Commerce to Static, so we could try using it
+to measure search performance.
+
+We didn't add an abstraction layer to GOV.UK Frontend Toolkit, because we don't
+know if there is a wider need for it, and we wanted to validate our use case with GOV.UK data.
+
+We send the search query as a custom dimension, because although Enhanced-Ecommerce
+can pick it up from the path ....it didn't work .... ?????
+
+For each "product" we send the content id and path, so we can uniquely identify
+each page. To avoid unnecessary data transfer when users are interacting with the page, we use Query Time Imports to add
+metadata to products later.
```

Comment:
Mention that GA has a hard limit of 8k and if the data packet is larger than this it will fail to send any page view data.  Further, we could mention that increases in the size of this data packet can affect page view data upload and data usage limit on people with slower connections. This is most relevant to people using a mobile device which overseas.